titration/utils/ui_state/ui_state.py

```diff
@@ -1,15 +1,17 @@
 class UIState:
-    def __init__(self, titrator):
+    def __init__(self, titrator, previousState=None):
         self.titrator = titrator
+        self.previousState = previousState
+        self.subState = 1
 
     def handleKey(self, key):
-        pass
+        raise Exception(self.__class__.__name__ + " requires a handleKey function")
 
     def name(self):
-        pass
+        raise Exception(self.__class__.__name__ + " requires a name function")
```

Comment:
In a future PR, you could change `raise Exception(self.__class__.__name__ + " requires a name function")` with ` return self.__class__.__name__`. Then remove `name()` from all subclasses.